fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java

```diff
@@ -1526,75 +1526,52 @@ private TableName qualifiedNameToTableName(QualifiedName qualifiedName) {
 
     private Type getType(StarRocksParser.TypeContext type) {
         if (type.baseType() != null) {
-            String signature = type.baseType().getText();
-            if (!type.typeParameter().isEmpty()) {
-                if (signature.equalsIgnoreCase("VARCHAR")) {
-                    if (type.typeParameter().size() > 1) {
-                        throw new SemanticException("VARCHAR can not contains multi type parameter");
-                    }
-
-                    if (type.typeParameter(0).INTEGER_VALUE() != null) {
-                        return ScalarType.createVarcharType(
-                                Integer.parseInt(type.typeParameter(0).INTEGER_VALUE().toString()));
-                    } else {
-                        throw new SemanticException("VARCHAR type parameter mush be integer");
-                    }
-                } else if (signature.equalsIgnoreCase("CHAR")) {
-                    if (type.typeParameter().size() > 1) {
-                        throw new SemanticException("CHAR can not contains multi type parameter");
-                    }
-                    if (type.typeParameter(0).INTEGER_VALUE() != null) {
-                        return ScalarType.createCharType(
-                                Integer.parseInt(type.typeParameter(0).INTEGER_VALUE().toString()));
-                    } else {
-                        throw new SemanticException("CHAR type parameter mush be integer");
-                    }
-                } else if (signature.equalsIgnoreCase("DECIMAL")) {
-                    throw new IllegalArgumentException("Unsupported type specification: " + type.getText());
-                }
-
-                throw new IllegalArgumentException("Unsupported type specification: " + type.getText());
-            }
-
-            if (signature.equalsIgnoreCase("BOOLEAN")) {
+            if (type.baseType().BOOLEAN() != null) {
```

Comment:
we can use switch here?  it looks so much if-else

Reply:
No, this is not an enumerable type, but whether the corresponding keyword is null